File: Kafka Ka Kalesh/d2.py
import sys

def solve():
    data = sys.stdin.read().strip().split()
    if not data:
        return
    it = iter(data)
    N = int(next(it))
    out = []
    for _ in range(N):
        P = int(next(it))
        C = int(next(it))
        S = int(next(it))
        L = int(next(it))

        produced = L // P

        attempts = 0
        if S <= L:
            attempts = (L - S) // C + 1

        successful = attempts
        if S < P and S <= L:
            successful = max(0, attempts - 1)

        lag = produced - successful
        out.append(str(lag))

    sys.stdout.write("\n".join(out) + "\n")

if __name__ == "__main__":
    solve()

File: Kafka Ka Kalesh/d2.cpp
#include <bits/stdc++.h>
using namespace std;

void solve() {
    long long P, C, S, L;
    cin >> P >> C >> S >> L;

    long long produced = L / P;

    long long attempts = 0;
    if (S <= L) {
        attempts = (L - S) / C + 1;
    }

    long long successful = attempts;
    if (S < P && S <= L) {
        successful = max(0LL, attempts - 1);
    }

    long long lag = produced - successful;
    cout << lag << "\n";
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int N;
    cin >> N;
    while (N--) {
        solve();
    }
    return 0;
}